okio-wasifilesystem/src/wasmMain/kotlin/WasiFileSystem.kt

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright (C) 2023 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import kotlin.wasm.unsafe.Pointer
+import kotlin.wasm.unsafe.UnsafeWasmMemoryApi
+import kotlin.wasm.unsafe.withScopedMemoryAllocator
+import okio.FileHandle
+import okio.FileMetadata
+import okio.FileSystem
+import okio.Path
+import okio.Sink
+import okio.Source
+import okio.internal.preview1.ErrnoException
+import okio.internal.preview1.FirstPreopenDirectoryTmp
+import okio.internal.preview1.dirnamelen
+import okio.internal.preview1.fd
+import okio.internal.preview1.fdClose
+import okio.internal.preview1.fd_readdir
+import okio.internal.preview1.oflag_creat
+import okio.internal.preview1.oflag_directory
+import okio.internal.preview1.oflag_excl
+import okio.internal.preview1.oflags
+import okio.internal.preview1.path_create_directory
+import okio.internal.preview1.path_open
+import okio.internal.preview1.readString
+import okio.internal.preview1.right_fd_readdir
+import okio.internal.preview1.right_fd_write
+import okio.internal.preview1.rights
+import okio.internal.preview1.write
+
+/**
+ * Use [WASI] to implement the Okio file system interface.
+ *
+ * [WASI]: https://wasi.dev/
+ */
+@OptIn(UnsafeWasmMemoryApi::class)
+object WasiFileSystem : FileSystem() {
```

Comment:
package declaration for this and `FileSink` is missing :)

Reply:
AKA they need to be moved to ./okio

Reply:
Whoops! Good catch.